lib/builders.nix

```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  pkgs,
+}: {
+  /*
+  Write a lua file to the nix store, formatted using stylua.
+
+  # Type
+
+  ```
+  writeLua :: String -> AttrSet -> String -> Derivation
+  ```
+
+  # Arguments
+
+  - [name] The name of the derivation
+  - [text] The content of the lua file
+  */
+  writeLua = name: text: let
+    raw = pkgs.writeText name text;
+    stylua = lib.getExe pkgs.stylua;
+  in
+    pkgs.runCommand name {} ''
+      ${stylua} \
+        --no-editorconfig \
+        --line-endings Unix \
+        --indent-type Spaces \
+        --indent-width 4 \
+        -- - < ${raw} 1> $out
```

Comment:
This might be more readable if we do something like

```shell
cp -TL ${raw} $out
${stylua} $out
```

`-- - < ${raw} 1> $out` is a little cryptic...

Reply:
Not a huge deal for me ?
@traxys any opinion ?